src/libc/sys/socket/shutdown_test.c

```diff
@@ -52,3 +54,61 @@ TEST(shutdown, example) {
   ASSERT_EQ(0, close(fds[0]));
   ASSERT_EQ(0, close(fds[1]));
 }
+
+static void *sleep_then_shutdown_read(void *arg) {
+  // Sleep for a second.
+  struct timespec ts = {.tv_sec = 0, .tv_nsec = 100000000L};
```

Comment:
Could you make this less than one second? The problem with such long sleeps is that they make the unit test suite a lot slower.   Even if we would sleep too briefly, the test should deal with that correctly, right?

Just use 100 msec. Also no need to put `L` behind the integer literal. :-)

Reply:
Oh, wait. You are already using 100 msec. It's just that the comment is stale.

Reply:
Ah, oops! Update coming. If we sleep too briefly, the `recv()` / `read()` may not have started blocking yet, but that may at most just cause the test to succeed by accident. With 100 ms, that would be insignificantly rare I think.